packages/go_router/analysis_options.yaml

```diff
@@ -1,5 +1,15 @@
+# This custom rule set only exists to allow very targeted changes
+# relative to the default repo settings, for specific use cases.
+# Please do NOT add more changes  here without consulting with
+# #hackers-ecosystem on Discord.
+
 include: ../../analysis_options.yaml
 
 analyzer:
   exclude:
+    # This directory deliberately has errors, to test `fix`.
     - "test_fixes/**"
+linter:
+  rules:
+    # Matches flutter/flutter, which disables this rule due to false positives.
+    unawaited_futures: false
```

Comment:
I am not too familiar with this lint. what's the false positives in this package?

Reply:
It looks like it's all in `test/`; there are about 40 violations, mostly `push` calls in this construction:

```
        goRouter.push(somePath);
        await tester.pumpAndSettle();
```

If you'd prefer to annotate all of the tests instead of opting out, that's definitely an options.

Reply:
(or do `// ignore_for_file: unawaited_futures` in all of the test files)

Reply:
sounds good, the pr is good as is, i can audit this later

Reply:
FWIW, the Navigator API (push/pop, etc) and the AnimationController API are the two main reasons why we don't have that lint enabled in the framework. It is actually completely safe to not await these calls because they will never throw. If that's also true for the goRouter API opting the entire package out of the lint seems a reasonable approach.

Reply:
Mild preference for ignore for file in the test than opting out the directory.

Reply:
> FWIW, the Navigator API (push/pop, etc) and the AnimationController API are the two main reasons why we don't have that lint enabled in the framework. It is actually completely safe to not await these calls because they will never throw.

Have we ever requested an annotation that would opt specific APIs out of this (a better named `@itsNormalAndFineToCallThisMethodWithoutAwait`) that would avoid us disabling a whole (useful) lint just for a handful of APIs?

Reply:
> Mild preference for ignore for file in the test than opting out the directory.

or we can have a analyzer_option.yaml in test folder

Reply:
> Have we ever requested an annotation that would opt specific APIs out of this (a better named @itsNormalAndFineToCallThisMethodWithoutAwait) that would avoid us disabling a whole (useful) lint just for a handful of APIs?

Yeah, there is a request for this on file with the linter project.